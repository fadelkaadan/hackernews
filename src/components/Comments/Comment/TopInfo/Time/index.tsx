import React from "react";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faClock } from "@fortawesome/free-regular-svg-icons";
import TimeAgo from "javascript-time-ago";
import en from "javascript-time-ago/locale/en";

interface TimeProps {
  time: number;
}

const Wrapper = styled.p`
  margin-right: 1rem;
`;

const Time = ({ time }: TimeProps) => {
  const covnvertUnixTime = () => {
    const unixTimestamp = time;
    const milliseconds = unixTimestamp * 1000;
    const dateObject = new Date(milliseconds);

    TimeAgo.addLocale(en);
    const timeAgo = new TimeAgo("en-US");
    return timeAgo.format(dateObject, "round");
  };

  return (
    <Wrapper>
      <FontAwesomeIcon icon={faClock} /> {covnvertUnixTime()}
    </Wrapper>
  );
};

export default Time;
