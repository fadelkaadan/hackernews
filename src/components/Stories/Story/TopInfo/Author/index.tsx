import React from "react";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUserCircle } from "@fortawesome/free-regular-svg-icons";

interface AuthorProps {
  name: string;
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  margin-right: 1rem;
`;

const AuthorName = styled.a`
  color: ${(props) => props.theme.main.text.secondary};
  text-decoration: none;
  margin-left: 0.5rem;
`;

const Author = ({ name }: AuthorProps) => {
  return (
    <Wrapper>
      <FontAwesomeIcon icon={faUserCircle} />
      <AuthorName href={"#"}>{name}</AuthorName>
    </Wrapper>
  );
};

export default Author;
