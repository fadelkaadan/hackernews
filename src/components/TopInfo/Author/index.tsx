import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUserCircle } from "@fortawesome/free-regular-svg-icons";

interface AuthorProps {
  name: string;
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  margin-right: 1rem;
`;

const AuthorName = styled(Link)`
  color: ${(props) => props.theme.main.text.secondary};
  text-decoration: none;
  margin-left: 0.5rem;
`;

const Author = ({ name }: AuthorProps) => {
  return (
    <Wrapper>
      <FontAwesomeIcon icon={faUserCircle} />
      <AuthorName to={`/users/${name}`}>{name}</AuthorName>
    </Wrapper>
  );
};

export default Author;
