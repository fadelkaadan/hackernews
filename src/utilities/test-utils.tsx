import React from "react";
import { Provider } from "react-redux";
import ThemeProvider from "../theme/ThemeProvider";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import { ThemeEnum } from "../store/preferences/types";
import { RootState } from "../store/rootReducer";

interface AllTheProvidersProps {
  children?: any;
}

const AllTheProviders = ({ children }: AllTheProvidersProps) => {
  const middlewares: any[] = [];
  const mockStore = configureStore(middlewares);

  const intialState: RootState = {
    preferences: {
      isPreferencesClicked: false,
      theme: ThemeEnum.NIGHT,
      isOneColumnLayout: true,
    },
    stories: {
      list: [],
      startAt: 0,
      pending: false,
      error: null,
    },
    user: {
      info: null,
      pending: false,
      error: null,

      stories: {
        list: [],
        startAt: 0,
        pending: false,
        error: null,
      },
    },
  };
  const store = mockStore(intialState);

  return (
    <Provider store={store}>
      <Router>
        <ThemeProvider>{children}</ThemeProvider>
      </Router>
    </Provider>
  );
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
