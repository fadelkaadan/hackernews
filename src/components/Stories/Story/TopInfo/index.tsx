import React from "react";
import styled from "styled-components";
import Score from "./Score";
import Time from "./Time";

interface TopInfoProps {
  time: number;
  score: number;
}

const Wrapper = styled.div`
  font-size: 12px;
  display: flex;
  align-items: center;
  color: ${(props) => props.theme.main.text.secondary};
`;

const TopInfo = ({ time, score }: TopInfoProps) => {
  return (
    <Wrapper>
      <Time time={time}/>
      <Score count={score} />
    </Wrapper>
  );
};

export default TopInfo;
