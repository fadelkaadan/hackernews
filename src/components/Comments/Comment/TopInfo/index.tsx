import React from "react";
import styled from "styled-components";
import Author from "./Author";
import Time from "./Time";

interface TopInfoProps {
  author: string;
  time: number;
}

const Wrapper = styled.div`
  font-size: 1.2rem;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  color: ${(props) => props.theme.main.text.secondary};
`;

const TopInfo = ({ author, time }: TopInfoProps) => {
  return (
    <Wrapper>
      <Author name={author} />
      <Time time={time} />
    </Wrapper>
  );
};

export default TopInfo;
