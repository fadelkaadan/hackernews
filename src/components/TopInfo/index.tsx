import React from "react";
import styled from "styled-components";
import Author from "./Author";
import Score from "./Score";
import Time from "./Time";

interface TopInfoProps {
  author: string;
  time: number;
  score?: number;
}

const Wrapper = styled.div`
  font-size: 1.2rem;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  color: ${(props) => props.theme.main.text.secondary};
`;

const TopInfo = ({ author, time, score }: TopInfoProps) => {
  return (
    <Wrapper>
      <Author name={author} />
      <Time time={time} />
      {score && <Score count={score} />}
    </Wrapper>
  );
};

export default TopInfo;
