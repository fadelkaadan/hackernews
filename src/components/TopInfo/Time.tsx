import React from "react";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faClock } from "@fortawesome/free-regular-svg-icons";
import TimeAgo from "javascript-time-ago";
import en from "javascript-time-ago/locale/en";
import { covnvertUnixTimeToDate } from "../../utilities/time";

interface TimeProps {
  time: number;
}

const Wrapper = styled.p`
  margin-right: 1rem;
`;

const Time = ({ time }: TimeProps) => {
  const getTimeAgo = () => {
    const dateObject = covnvertUnixTimeToDate(time);
    TimeAgo.addLocale(en);
    const timeAgo = new TimeAgo("en-US");
    return timeAgo.format(dateObject, "round");
  };

  return (
    <Wrapper>
      <FontAwesomeIcon icon={faClock} /> {getTimeAgo()}
    </Wrapper>
  );
};

export default Time;
