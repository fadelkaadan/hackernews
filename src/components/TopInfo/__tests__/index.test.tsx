import React from "react";
import { render, screen, RenderResult } from "../../../utilities/test-utils";
import TopInfo from "../";

describe("TopInfo", () => {
  const topInfo = {
    author: "fadel",
    time: 1587682800,
    score: 123,
  };

  it("Should match the snapshot", () => {
    const { container }: RenderResult = render(
      <TopInfo
        author={topInfo.author}
        time={topInfo.time}
        score={topInfo.score}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should render top info", () => {
    render(
      <TopInfo
        author={topInfo.author}
        time={topInfo.time}
        score={topInfo.score}
      />
    );
    const time: HTMLElement = screen.getByText(/ago/i);
    expect(time).toBeInTheDocument();

    const author: HTMLElement = screen.getByRole("link", {
      name: topInfo.author,
    });
    expect(author).toBeInTheDocument();

    const score: HTMLElement = screen.getByText(`${topInfo.score}`);
    expect(score).toBeInTheDocument();
  });
});
