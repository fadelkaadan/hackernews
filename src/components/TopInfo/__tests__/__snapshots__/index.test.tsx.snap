// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopInfo Should match the snapshot 1`] = `
<div
  class="sc-eCApGN eAINuR"
>
  <div
    class="sc-bdnylx gSfPKd"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-user-circle fa-w-16 "
      data-icon="user-circle"
      data-prefix="far"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
        fill="currentColor"
      />
    </svg>
    <a
      class="sc-gtssRu hIgpwU"
      href="/users/fadel"
    >
      fadel
    </a>
  </div>
  <p
    class="sc-hKFyIo daGSDy"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-clock fa-w-16 "
      data-icon="clock"
      data-prefix="far"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
        fill="currentColor"
      />
    </svg>
     
    9 months ago
  </p>
  <p
    class="sc-dlnjPT bxfkjj"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-arrow-alt-circle-up fa-w-16 "
      data-icon="arrow-alt-circle-up"
      data-prefix="far"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 504c137 0 248-111 248-248S393 8 256 8 8 119 8 256s111 248 248 248zm0-448c110.5 0 200 89.5 200 200s-89.5 200-200 200S56 366.5 56 256 145.5 56 256 56zm20 328h-40c-6.6 0-12-5.4-12-12V256h-67c-10.7 0-16-12.9-8.5-20.5l99-99c4.7-4.7 12.3-4.7 17 0l99 99c7.6 7.6 2.2 20.5-8.5 20.5h-67v116c0 6.6-5.4 12-12 12z"
        fill="currentColor"
      />
    </svg>
     
    123
  </p>
</div>
`;
